// bear-react-dropdown/dist/bear-react-dropdown.es.js
import Le, { useState as We, useRef as te, useEffect as Ye, useCallback as K, startTransition as Ue } from "react";
var d = {}, xr = {
  get exports() {
    return d;
  },
  set exports(u) {
    d = u;
  }
}, H = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Fe;
function br() {
  if (Fe)
    return H;
  Fe = 1;
  var u = Le, x = Symbol.for("react.element"), v = Symbol.for("react.fragment"), c = Object.prototype.hasOwnProperty, R = u.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, g = { key: !0, ref: !0, __self: !0, __source: !0 };
  function T(C, _, $) {
    var E, O = {}, P = null, S = null;
    $ !== void 0 && (P = "" + $), _.key !== void 0 && (P = "" + _.key), _.ref !== void 0 && (S = _.ref);
    for (E in _)
      c.call(_, E) && !g.hasOwnProperty(E) && (O[E] = _[E]);
    if (C && C.defaultProps)
      for (E in _ = C.defaultProps, _)
        O[E] === void 0 && (O[E] = _[E]);
    return { $$typeof: x, type: C, key: P, ref: S, props: O, _owner: R.current };
  }
  return H.Fragment = v, H.jsx = T, H.jsxs = T, H;
}
var z = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var De;
function yr() {
  return De || (De = 1, process.env.NODE_ENV !== "production" && function() {
    var u = Le, x = Symbol.for("react.element"), v = Symbol.for("react.portal"), c = Symbol.for("react.fragment"), R = Symbol.for("react.strict_mode"), g = Symbol.for("react.profiler"), T = Symbol.for("react.provider"), C = Symbol.for("react.context"), _ = Symbol.for("react.forward_ref"), $ = Symbol.for("react.suspense"), E = Symbol.for("react.suspense_list"), O = Symbol.for("react.memo"), P = Symbol.for("react.lazy"), S = Symbol.for("react.offscreen"), D = Symbol.iterator, G = "@@iterator";
    function L(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = D && e[D] || e[G];
      return typeof r == "function" ? r : null;
    }
    var N = u.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function a(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        s("error", e, t);
      }
    }
    function s(e, r, t) {
      {
        var n = N.ReactDebugCurrentFrame, f = n.getStackAddendum();
        f !== "" && (r += "%s", t = t.concat([f]));
        var m = t.map(function(o) {
          return String(o);
        });
        m.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, m);
      }
    }
    var l = !1, h = !1, w = !1, Be = !1, Ge = !1, de;
    de = Symbol.for("react.module.reference");
    function Je(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === c || e === g || Ge || e === R || e === $ || e === E || Be || e === S || l || h || w || typeof e == "object" && e !== null && (e.$$typeof === P || e.$$typeof === O || e.$$typeof === T || e.$$typeof === C || e.$$typeof === _ || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === de || e.getModuleId !== void 0));
    }
    function qe(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var f = r.displayName || r.name || "";
      return f !== "" ? t + "(" + f + ")" : t;
    }
    function ve(e) {
      return e.displayName || "Context";
    }
    function A(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && a("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case c:
          return "Fragment";
        case v:
          return "Portal";
        case g:
          return "Profiler";
        case R:
          return "StrictMode";
        case $:
          return "Suspense";
        case E:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case C:
            var r = e;
            return ve(r) + ".Consumer";
          case T:
            var t = e;
            return ve(t._context) + ".Provider";
          case _:
            return qe(e, e.render, "ForwardRef");
          case O:
            var n = e.displayName || null;
            return n !== null ? n : A(e.type) || "Memo";
          case P: {
            var f = e, m = f._payload, o = f._init;
            try {
              return A(o(m));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var W = Object.assign, J = 0, me, pe, he, ge, _e, xe, be;
    function ye() {
    }
    ye.__reactDisabledLog = !0;
    function He() {
      {
        if (J === 0) {
          me = console.log, pe = console.info, he = console.warn, ge = console.error, _e = console.group, xe = console.groupCollapsed, be = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: ye,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        J++;
      }
    }
    function ze() {
      {
        if (J--, J === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: W({}, e, {
              value: me
            }),
            info: W({}, e, {
              value: pe
            }),
            warn: W({}, e, {
              value: he
            }),
            error: W({}, e, {
              value: ge
            }),
            group: W({}, e, {
              value: _e
            }),
            groupCollapsed: W({}, e, {
              value: xe
            }),
            groupEnd: W({}, e, {
              value: be
            })
          });
        }
        J < 0 && a("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var ne = N.ReactCurrentDispatcher, ae;
    function X(e, r, t) {
      {
        if (ae === void 0)
          try {
            throw Error();
          } catch (f) {
            var n = f.stack.trim().match(/\n( *(at )?)/);
            ae = n && n[1] || "";
          }
        return `
` + ae + e;
      }
    }
    var ie = !1, Z;
    {
      var Ke = typeof WeakMap == "function" ? WeakMap : Map;
      Z = new Ke();
    }
    function Ee(e, r) {
      if (!e || ie)
        return "";
      {
        var t = Z.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      ie = !0;
      var f = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var m;
      m = ne.current, ne.current = null, He();
      try {
        if (r) {
          var o = function() {
            throw Error();
          };
          if (Object.defineProperty(o.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(o, []);
            } catch (F) {
              n = F;
            }
            Reflect.construct(e, [], o);
          } else {
            try {
              o.call();
            } catch (F) {
              n = F;
            }
            e.call(o.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (F) {
            n = F;
          }
          e();
        }
      } catch (F) {
        if (F && n && typeof F.stack == "string") {
          for (var i = F.stack.split(`
`), j = n.stack.split(`
`), b = i.length - 1, y = j.length - 1; b >= 1 && y >= 0 && i[b] !== j[y]; )
            y--;
          for (; b >= 1 && y >= 0; b--, y--)
            if (i[b] !== j[y]) {
              if (b !== 1 || y !== 1)
                do
                  if (b--, y--, y < 0 || i[b] !== j[y]) {
                    var I = `
` + i[b].replace(" at new ", " at ");
                    return e.displayName && I.includes("<anonymous>") && (I = I.replace("<anonymous>", e.displayName)), typeof e == "function" && Z.set(e, I), I;
                  }
                while (b >= 1 && y >= 0);
              break;
            }
        }
      } finally {
        ie = !1, ne.current = m, ze(), Error.prepareStackTrace = f;
      }
      var U = e ? e.displayName || e.name : "", Ae = U ? X(U) : "";
      return typeof e == "function" && Z.set(e, Ae), Ae;
    }
    function Xe(e, r, t) {
      return Ee(e, !1);
    }
    function Ze(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function Q(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return Ee(e, Ze(e));
      if (typeof e == "string")
        return X(e);
      switch (e) {
        case $:
          return X("Suspense");
        case E:
          return X("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case _:
            return Xe(e.render);
          case O:
            return Q(e.type, r, t);
          case P: {
            var n = e, f = n._payload, m = n._init;
            try {
              return Q(m(f), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var ee = Object.prototype.hasOwnProperty, Re = {}, Ce = N.ReactDebugCurrentFrame;
    function re(e) {
      if (e) {
        var r = e._owner, t = Q(e.type, e._source, r ? r.type : null);
        Ce.setExtraStackFrame(t);
      } else
        Ce.setExtraStackFrame(null);
    }
    function Qe(e, r, t, n, f) {
      {
        var m = Function.call.bind(ee);
        for (var o in e)
          if (m(e, o)) {
            var i = void 0;
            try {
              if (typeof e[o] != "function") {
                var j = Error((n || "React class") + ": " + t + " type `" + o + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[o] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw j.name = "Invariant Violation", j;
              }
              i = e[o](r, o, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (b) {
              i = b;
            }
            i && !(i instanceof Error) && (re(f), a("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, o, typeof i), re(null)), i instanceof Error && !(i.message in Re) && (Re[i.message] = !0, re(f), a("Failed %s type: %s", t, i.message), re(null));
          }
      }
    }
    var er = Array.isArray;
    function se(e) {
      return er(e);
    }
    function rr(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function tr(e) {
      try {
        return je(e), !1;
      } catch {
        return !0;
      }
    }
    function je(e) {
      return "" + e;
    }
    function Te(e) {
      if (tr(e))
        return a("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", rr(e)), je(e);
    }
    var q = N.ReactCurrentOwner, nr = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Oe, Pe, le;
    le = {};
    function ar(e) {
      if (ee.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function ir(e) {
      if (ee.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function sr(e, r) {
      if (typeof e.ref == "string" && q.current && r && q.current.stateNode !== r) {
        var t = A(q.current.type);
        le[t] || (a('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', A(q.current.type), e.ref), le[t] = !0);
      }
    }
    function lr(e, r) {
      {
        var t = function() {
          Oe || (Oe = !0, a("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function or(e, r) {
      {
        var t = function() {
          Pe || (Pe = !0, a("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var ur = function(e, r, t, n, f, m, o) {
      var i = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: x,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: o,
        // Record the component responsible for creating this element.
        _owner: m
      };
      return i._store = {}, Object.defineProperty(i._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(i, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(i, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: f
      }), Object.freeze && (Object.freeze(i.props), Object.freeze(i)), i;
    };
    function cr(e, r, t, n, f) {
      {
        var m, o = {}, i = null, j = null;
        t !== void 0 && (Te(t), i = "" + t), ir(r) && (Te(r.key), i = "" + r.key), ar(r) && (j = r.ref, sr(r, f));
        for (m in r)
          ee.call(r, m) && !nr.hasOwnProperty(m) && (o[m] = r[m]);
        if (e && e.defaultProps) {
          var b = e.defaultProps;
          for (m in b)
            o[m] === void 0 && (o[m] = b[m]);
        }
        if (i || j) {
          var y = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          i && lr(o, y), j && or(o, y);
        }
        return ur(e, i, j, f, n, q.current, o);
      }
    }
    var oe = N.ReactCurrentOwner, Se = N.ReactDebugCurrentFrame;
    function Y(e) {
      if (e) {
        var r = e._owner, t = Q(e.type, e._source, r ? r.type : null);
        Se.setExtraStackFrame(t);
      } else
        Se.setExtraStackFrame(null);
    }
    var ue;
    ue = !1;
    function ce(e) {
      return typeof e == "object" && e !== null && e.$$typeof === x;
    }
    function Ie() {
      {
        if (oe.current) {
          var e = A(oe.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function fr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Ne = {};
    function dr(e) {
      {
        var r = Ie();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function ke(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = dr(r);
        if (Ne[t])
          return;
        Ne[t] = !0;
        var n = "";
        e && e._owner && e._owner !== oe.current && (n = " It was passed a child from " + A(e._owner.type) + "."), Y(e), a('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), Y(null);
      }
    }
    function $e(e, r) {
      {
        if (typeof e != "object")
          return;
        if (se(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            ce(n) && ke(n, r);
          }
        else if (ce(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var f = L(e);
          if (typeof f == "function" && f !== e.entries)
            for (var m = f.call(e), o; !(o = m.next()).done; )
              ce(o.value) && ke(o.value, r);
        }
      }
    }
    function vr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === _ || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === O))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = A(r);
          Qe(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !ue) {
          ue = !0;
          var f = A(r);
          a("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", f || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && a("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function mr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            Y(e), a("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), Y(null);
            break;
          }
        }
        e.ref !== null && (Y(e), a("Invalid attribute `ref` supplied to `React.Fragment`."), Y(null));
      }
    }
    function we(e, r, t, n, f, m) {
      {
        var o = Je(e);
        if (!o) {
          var i = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (i += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var j = fr(f);
          j ? i += j : i += Ie();
          var b;
          e === null ? b = "null" : se(e) ? b = "array" : e !== void 0 && e.$$typeof === x ? (b = "<" + (A(e.type) || "Unknown") + " />", i = " Did you accidentally export a JSX literal instead of a component?") : b = typeof e, a("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", b, i);
        }
        var y = cr(e, r, t, f, m);
        if (y == null)
          return y;
        if (o) {
          var I = r.children;
          if (I !== void 0)
            if (n)
              if (se(I)) {
                for (var U = 0; U < I.length; U++)
                  $e(I[U], e);
                Object.freeze && Object.freeze(I);
              } else
                a("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              $e(I, e);
        }
        return e === c ? mr(y) : vr(y), y;
      }
    }
    function pr(e, r, t) {
      return we(e, r, t, !0);
    }
    function hr(e, r, t) {
      return we(e, r, t, !1);
    }
    var gr = hr, _r = pr;
    z.Fragment = c, z.jsx = gr, z.jsxs = _r;
  }()), z;
}
(function(u) {
  process.env.NODE_ENV === "production" ? u.exports = br() : u.exports = yr();
})(xr);
const k = {
  elCLassPrefix: "bear-react-dropdown"
}, p = {
  root: `${k.elCLassPrefix}__root`,
  textField: `${k.elCLassPrefix}__text-field`,
  list: `${k.elCLassPrefix}__list`,
  listItem: `${k.elCLassPrefix}__list-item`,
  listItemActive: `${k.elCLassPrefix}__list-item-active`,
  listItemText: `${k.elCLassPrefix}__list-item-text`,
  listItemTextPlaceholder: `${k.elCLassPrefix}__list-item-text-placeholder`,
  listItemAvatar: `${k.elCLassPrefix}__list-item-avatar`,
  listItemChecked: `${k.elCLassPrefix}__list-item-checked`,
  listGroupName: `${k.elCLassPrefix}__list-group-name`,
  listGroupChildren: `${k.elCLassPrefix}__list-group-children`
};
var fe = {}, Er = {
  get exports() {
    return fe;
  },
  set exports(u) {
    fe = u;
  }
};
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
(function(u) {
  (function() {
    var x = {}.hasOwnProperty;
    function v() {
      for (var c = [], R = 0; R < arguments.length; R++) {
        var g = arguments[R];
        if (g) {
          var T = typeof g;
          if (T === "string" || T === "number")
            c.push(g);
          else if (Array.isArray(g)) {
            if (g.length) {
              var C = v.apply(null, g);
              C && c.push(C);
            }
          } else if (T === "object") {
            if (g.toString !== Object.prototype.toString && !g.toString.toString().includes("[native code]")) {
              c.push(g.toString());
              continue;
            }
            for (var _ in g)
              x.call(g, _) && g[_] && c.push(_);
          }
        }
      }
      return c.join(" ");
    }
    u.exports ? (v.default = v, u.exports = v) : window.classNames = v;
  })();
})(Er);
const M = fe, V = (u) => u.groupName !== void 0, B = (u, x) => (x == null ? void 0 : x.length) > 0 ? u.filter((v) => String(v.text).toLowerCase().indexOf(x.toLowerCase()) !== -1) : u;
function Me(u) {
  return u == null || typeof u == "string" && u.trim().length === 0;
}
function Rr(u, x) {
  return x === -1 || x > u.length - 1 ? u : [...u.slice(0, x), ...u.slice(x + 1)];
}
const Ve = () => /* @__PURE__ */ d.jsx("svg", { viewBox: "0 0 1024 1024", version: "1.1", xmlns: "http://www.w3.org/2000/svg", width: "15", height: "15", children: /* @__PURE__ */ d.jsx("path", { d: "M486.4 661.333333l273.066667-273.066666c17.066667-17.066667 17.066667-42.666667 0-59.733334-17.066667-17.066667-42.666667-17.066667-59.733334 0L426.666667 601.6l-102.4-102.4c-17.066667-17.066667-42.666667-17.066667-59.733334 0-17.066667 17.066667-17.066667 42.666667 0 59.733333l102.4 102.4 29.866667 29.866667c17.066667 17.066667 42.666667 17.066667 59.733333 0l29.866667-29.866667z" }) }), Cr = 30, jr = 15, Tr = 30 * jr / 2, Nr = ({
  className: u,
  style: x,
  options: v,
  value: c,
  onChange: R,
  onClick: g,
  searchTextPlaceholder: T = "type keyword...",
  isSearchEnable: C = !1,
  isCheckedEnable: _ = !0,
  isAvatarEnable: $ = !1,
  isDark: E = !1
}) => {
  const [O, P] = We(""), S = te(null), D = te(null);
  Ye(() => {
    var s;
    if (C && (S == null ? void 0 : S.current) !== null && S.current.focus(), D.current && !Me(c)) {
      const l = (v == null ? void 0 : v.findIndex((h) => V(h) ? h.children.findIndex((w) => w.value === c) : h.value === c)) ?? -1;
      l >= 0 && ((s = D.current) == null || s.scrollTo({ top: l * Cr - Tr }));
    }
  }, []);
  const G = K((s) => {
    Ue(() => {
      P(s.target.value);
    });
  }, []), L = K((s) => {
    R && c !== s && R(s), g && g(s);
  }, [R, g, c]), N = (s) => {
    const l = c === s.value;
    return /* @__PURE__ */ d.jsxs(
      "button",
      {
        type: "button",
        className: M(p.listItem, { [p.listItemActive]: l }),
        onClick: () => L(s.value),
        children: [
          _ && /* @__PURE__ */ d.jsx("div", { className: p.listItemChecked, children: l && /* @__PURE__ */ d.jsx(Ve, {}) }),
          $ && /* @__PURE__ */ d.jsx("div", { className: p.listItemAvatar, style: s.avatarUrl ? { backgroundImage: `url(${s.avatarUrl})` } : {} }),
          /* @__PURE__ */ d.jsx("div", { className: M(p.listItemText, { [p.listItemTextPlaceholder]: s.value === "" }), children: s.text })
        ]
      },
      `option-${s.value}`
    );
  }, a = K((s) => {
    const l = v == null ? void 0 : v.filter((h) => V(h) ? B(h.children.map((w) => w), s).length > 0 : B([h], s).length > 0).map((h) => V(h) ? /* @__PURE__ */ d.jsxs("div", { children: [
      /* @__PURE__ */ d.jsx("div", { className: p.listGroupName, children: h.groupName }),
      /* @__PURE__ */ d.jsx("div", { className: p.listGroupChildren, children: B(h.children, s).map(
        (w) => N(w)
      ) })
    ] }, `group_${h.groupName}`) : N(h));
    return l && l.length > 0 ? l : /* @__PURE__ */ d.jsx(
      "div",
      {
        className: p.listItem,
        onClick: () => L(null),
        children: /* @__PURE__ */ d.jsx("div", { className: p.listItemText, children: "No data" })
      },
      "no-data"
    );
  }, [v, c]);
  return /* @__PURE__ */ d.jsxs("div", { className: M(p.root, u, { "dark-theme": E }), style: x, children: [
    C && /* @__PURE__ */ d.jsx(
      "input",
      {
        className: p.textField,
        type: "text",
        value: O,
        onChange: G,
        placeholder: T
      }
    ),
    /* @__PURE__ */ d.jsx("div", { className: p.list, ref: D, children: a(O) })
  ] });
}, Or = 30, Pr = 15, Sr = 30 * Pr / 2, kr = ({
  className: u,
  style: x,
  options: v,
  value: c,
  onChange: R,
  searchTextPlaceholder: g = "type keyword...",
  isSearchEnable: T = !1,
  isCheckedEnable: C = !0,
  isAvatarEnable: _ = !1,
  isDark: $ = !1
}) => {
  const [E, O] = We(""), P = te(null), S = te(null);
  Ye(() => {
    var a;
    if (T && (P == null ? void 0 : P.current) !== null && P.current.focus(), S.current && !Me(c)) {
      const s = (v == null ? void 0 : v.findIndex((l) => V(l) ? l.children.findIndex((h) => h.value === c) : l.value === c)) ?? -1;
      s >= 0 && ((a = S.current) == null || a.scrollTo({ top: s * Or - Sr }));
    }
  }, []);
  const D = K((a) => {
    Ue(() => {
      O(a.target.value);
    });
  }, []), G = (a) => {
    if (R) {
      const s = (c == null ? void 0 : c.findIndex((w) => w === a)) ?? -1;
      let l = null;
      const h = c ?? [];
      s >= 0 ? l = Rr(h, s) : l = [...h, a], (l == null ? void 0 : l.length) === 0 && (l = null), JSON.stringify(l) !== JSON.stringify(c) && R(l);
    }
  }, L = (a) => {
    const s = (c == null ? void 0 : c.includes(a.value)) ?? !1;
    return /* @__PURE__ */ d.jsxs(
      "button",
      {
        type: "button",
        className: M(p.listItem, { [p.listItemActive]: s }),
        onClick: () => G(a.value),
        children: [
          C && /* @__PURE__ */ d.jsx("div", { className: p.listItemChecked, children: s && /* @__PURE__ */ d.jsx(Ve, {}) }),
          _ && /* @__PURE__ */ d.jsx("div", { className: p.listItemAvatar, style: a.avatarUrl ? { backgroundImage: `url(${a.avatarUrl})` } : {} }),
          /* @__PURE__ */ d.jsx("div", { className: M(p.listItemText, { [p.listItemTextPlaceholder]: a.value === "" }), children: a.text })
        ]
      },
      `option-${a.value}`
    );
  }, N = K((a) => {
    const s = v == null ? void 0 : v.filter((l) => V(l) ? B(l.children.map((h) => h), a).length > 0 : B([l], a).length > 0).map((l) => V(l) ? /* @__PURE__ */ d.jsxs("div", { children: [
      /* @__PURE__ */ d.jsx("div", { className: p.listGroupName, children: l.groupName }),
      /* @__PURE__ */ d.jsx("div", { className: p.listGroupChildren, children: B(l.children, a).map((h) => L(h)) })
    ] }, `group_${l.groupName}`) : L(l));
    return s && s.length > 0 ? s : /* @__PURE__ */ d.jsx(
      "div",
      {
        className: p.listItem,
        onClick: () => {
        },
        children: /* @__PURE__ */ d.jsx("div", { className: p.listItemText, children: "No data" })
      },
      "no-data"
    );
  }, [v]);
  return /* @__PURE__ */ d.jsxs("div", { className: M(p.root, u, { "dark-theme": $ }), style: x, children: [
    T && /* @__PURE__ */ d.jsx(
      "input",
      {
        className: p.textField,
        type: "text",
        value: E,
        onChange: D,
        placeholder: g
      }
    ),
    /* @__PURE__ */ d.jsx("div", { className: p.list, ref: S, children: N(E) })
  ] });
};
export {
  Nr as Dropdown,
  p as DropdownElClassNames,
  kr as DropdownMulti,
  B as filterOptions,
  Me as isEmpty,
  V as isGroupOptions,
  Rr as removeByIndex
};
